File: Iniciante/URI 1040.c
 #include <stdio.h>

 int main () {

    double N1, N2, N3, N4, NE, media;
    scanf ("%lf%lf%lf%lf", &N1, &N2, &N3, &N4);
    media = (((N1 * 2) + (N2 * 3) + (N3 * 4) + (N4 * 1))/10);
    printf("Media: %.1lf\n", media);
    if (media >= 7.0){
            printf ("Aluno aprovado.\n");
    }

    else if (media >= 5.0){
            
            printf ("Aluno em exame.\n");
            scanf ("%lf", &NE);
            printf ("Nota do exame: %.1lf\n", NE);
            if (NE + media / 2.0 > 5.0){
            printf ("Aluno aprovado.\n");
    }

     else {

         printf ("Aluno reprovado.\n") ;

        }

   printf("Media final: %.1f\n", (NE + media ) / 2.0);

    }

    else {

        printf ("Aluno reprovado.\n");

    }
    return 0;
}

File: Iniciante/URI 1040.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

/**
 * Escreva a sua solução aqui
 * Code your solution here
 * Escriba su solución aquí
 */
let linhas = lines.shift().split(' ')
let notaA = parseFloat(linhas[0])
let notaB = parseFloat(linhas[1])
let notaC = parseFloat(linhas[2])
let notaD = parseFloat(linhas[3])
let media = ((notaA * 2) + (notaB * 3) + (notaC * 4) + (notaD * 1)) / 10
console.log("Media: "+media.toFixed(1))
if (media >= 7) {
    console.log("Aluno aprovado.")
} else if (media < 7 && media >= 5) {
    let exame = parseFloat(lines.shift())
    media = (media + exame) / 2
    console.log("Aluno em exame.")
    console.log("Nota do exame: "+exame.toFixed(1))
    if (media >= 5.0) {
        console.log("Aluno aprovado.")
    } else {
        console.log("Aluno reprovado.")
    }
    console.log("Media final: "+ media.toFixed(1))
} else {
    console.log("Aluno reprovado.")
}